05A-foundational-skills-DRAFT.Rmd

```diff
@@ -41,24 +41,471 @@ Make an illustration that shows how these are related, highlighting where in the
+Seeing the `+` sign in the Console
+Fixing it by pressing `Esc`
+
+## Finicky functions (rename this section)
+
+```{r}
+ma_data_init %>%
+    group_by(`District Name`) %>%
+    count() %>%
+    filter(n > 10) %>%
+    arrange(desc(n))
+
+ma_data_init %>%
+    group_by(`District Name`) %>%
+    count() %>%
+    filter(n = 10) %>%
+    arrange(desc(n))
+
+ma_data_init %>%
+    group_by(`District Name`) %>%
+    count() %>%
+    filter(n == 10) %>%
+    arrange(desc(n))
+
+ma_data_init %>% 
+    rename(district_name = `District Name`,
+           grade = Grade) %>% 
+    select(district_name, grade)
+```
+
+### When functions need arguments and when they don't
+(Is there any way to know for sure?) 
+Help documentation 
+
+### Functions within functions
+Saw in import as well
+
+### The difference between `=` and `==` 
+Assignment vs. equality
+
+### Writing your own functions
+Basic function structure
+Quosures - out of scope of this chapter
+
+## SECTION ON c() and %in%
+
+```{r}
+
+```
+
+
+## Creating objects
+In R, everything is an object 
+```{r}
+ma_data %>% 
+    clean_names()
+
+01_ma_data <- ma_data_init %>%  #' intentional error
+    clean_names()
+
+$_ma_data <- ma_data_init %>%  #' intentional error
+    clean_names()
+
+ma_data_01 <- ma_data_init %>% 
+    clean_names()
+```
+
+### Object-oriented programming (OOP)  
+How in-depth should we go? May not be relevant
+
+### Analysis vs. development  
+Using pre-made functions vs. writing our own functions and packages, exploring pre-made functions 
+
+
+
+
+
+
+## Old text - keep until reincorporated//readdressed
+To break that down another way, it's helpful to know that R reads everything on the left of the assignment operator, `<-`, first, and while it reads from left to right, it starts with the innermost set of parentheses per object.  
+
+We can recognize functions in R by the use of parentheses. For example, `data()` is a function that pulls up all of the existing data sets within R.  
+The parentheses within a function can be empty, but sometimes a function requires arguments in order to work. Arguments are the components that go inside the parentheses of a function, like when we coerced our `UCBAdmissions` data into a tibble.  
+
+We can see which arguments a function uses by calling up the help documentation. We do this by going to the Console and typing a `?` followed by the function's name, then hitting enter. For example, run `?data` in the Console (notice that there is not a space between the question mark and the function's name).  
+
+Now try `?mutate` - what's different?   
+
+`mutate` is a function from the `dplyr` package - a package we'll use extensively throughout this book. But because we haven't brought the `dplyr` package into R yet (how does this fit into the mental model? Is this the right place to talk about this? - might be too far out of scope and belong at the end, in the explanations section.) R doesn't "see" the `mutate` function as existing.  
```

Comment:
This is a great question! I think I'd err on the side of simpler here. If we're looking to serve up frequent wins, we can show them how to load `tidyverse` and save a more in-depth discussion of `tidyverse` as a suite of packages for later. It feels like this chapter is aimed at getting the reader ready to do the walkthroughs. There are so many little wins we can get for them just within that goal, which I think will be important this early in the book.